release_docs/RELEASE.txt

```diff
@@ -47,6 +47,24 @@ New Features
 
     Configuration:
     -------------
+    - Added signed Windows msi binary and signed Apple dmg binary files.
+
+      The release process now provides signed Windows and Apple installation
+      binaries in addition to the debian and rpm installation binaries. Also
+      these installer files are no longer compressed into packaged archives.
+
+    - Added configuration option for internal threading/concurrency support:
+
+          CMake:      HDF5_ENABLE_THREADS (ON/OFF) (Default: ON)
+          Autotools:  --enable-threads    (yes/no) (Default: yes)
+
+      This option enables support for threading and concurrency algorithms
+      within the HDF5 library.  It is required for, but separate from, the
+      'threadsafe' configure option, which makes the HDF5 API safe to call from
+      multiple threads.  It is possible to enable the 'threads' option and
+      disable the 'threadsafe' option, but not vice versa.  The 'threads' option
+      must be on to enable the subfiling VFD.
+
```

Comment:
This should not have been merged to 1.14. --enable-threads and HDF5_ENABLE_THREADS should be develop-only.